packages/kotti-ui/source/kotti-field-file-upload/validators.ts

```diff
@@ -0,0 +1,50 @@
+import { KottiFieldFileUpload } from './types'
+
+export const isSelectingMultipleFilesWhenNotAllowed = (
+	allowMultiple: boolean,
+	selectionSize: number,
+): boolean => !allowMultiple && selectionSize > 1
+
+const isValidFileSize = (fileSize: number, maxFileSize: number): boolean =>
+	fileSize < maxFileSize
+
+const isValidFileExtension = (
+	fileName: string,
+	validFileExtensions: Array<string>,
+): boolean => {
+	if (validFileExtensions.length === 0) return true
+
+	const fileExtension = (fileName.split('.').pop() ?? '').trim().toUpperCase()
+
+	if (!fileExtension) return false
+
+	return validFileExtensions
+		.map((extension) => extension.trim().toUpperCase())
+		.includes(fileExtension)
+}
+
+export const isValidUploadedFileStatus = (
+	valueItem: KottiFieldFileUpload.ValueInternal[0],
+) =>
+	valueItem.status === KottiFieldFileUpload.Status.Common.INVALID ||
+	valueItem.status === KottiFieldFileUpload.Status.Common.UPLOADED
+
+export const validateFile = ({
+	extensions,
+	fileName,
+	fileSize,
+	maxFileSize,
+}: {
+	extensions: KottiFieldFileUpload.PropsInternal['extensions']
+	fileName: string
+	fileSize: number
+	maxFileSize: KottiFieldFileUpload.PropsInternal['maxFileSize']
+}): KottiFieldFileUpload.Validation => {
+	if (!isValidFileExtension(fileName, extensions))
+		return KottiFieldFileUpload.Validation.INVALID_EXTENSION
```

Comment:
Question:

If both fail (max size and file extension), you will give priority to the Invalid Extension error first

Aka you don't show an array, you show the first error first then I'd assume you revalidate

Why don't you return an array of all errors - 

Have you aligned with alberto on this

Reply:
Yes, Alberto validated that file extension error has priority.